controller_manager/src/resource_manager.cpp

```diff
@@ -234,17 +288,47 @@ bool ResourceManager::command_interface_exists(const std::string & key) const
          resource_storage_->command_interface_map_.end();
 }
 
-size_t ResourceManager::actuator_interfaces_size() const
+void ResourceManager::import_component(
+  std::unique_ptr<hardware_interface::components::ActuatorInterface> actuator)
+{
+  resource_storage_->actuators_.emplace_back(
+    hardware_interface::components::Actuator(
+      std::move(
+        actuator)));
+  resource_storage_->import_state_interfaces(resource_storage_->actuators_.back());
+  resource_storage_->import_command_interfaces(
+    resource_storage_->actuators_.back(), claimed_command_interface_map_);
+}
+
+size_t ResourceManager::actuator_components_size() const
 {
   return resource_storage_->actuators_.size();
 }
 
-size_t ResourceManager::sensor_interfaces_size() const
+void ResourceManager::import_component(
+  std::unique_ptr<hardware_interface::components::SensorInterface> sensor)
+{
+  resource_storage_->sensors_.emplace_back(
+    hardware_interface::components::Sensor(std::move(sensor)));
+  resource_storage_->import_state_interfaces(resource_storage_->sensors_.back());
+}
+
+size_t ResourceManager::sensor_components_size() const
 {
   return resource_storage_->sensors_.size();
 }
 
-size_t ResourceManager::system_interfaces_size() const
+void ResourceManager::import_component(
+  std::unique_ptr<hardware_interface::components::SystemInterface> system)
```

Comment:
After seeing hardware_interface::components::SystemInterface et al a couple of times I think that the components middle namespace is soon going to outlive it's usefulness, we should aim to remove it & perhaps rename the hardware_interface package.

Reply:
see #230